pkg/controller/installation/core_controller.go

```diff
@@ -1270,18 +1275,31 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	}
 
 	// Create a component handler to create or update the rendered components.
-	handler := utils.NewComponentHandler(log, r.client, r.scheme, instance)
+	compHandler := utils.NewComponentHandler(log, r.client, r.scheme, instance)
+
+	// Most of the components need only the standard processing.
 	for _, component := range components {
-		if err := handler.CreateOrUpdateOrDelete(ctx, component, nil); err != nil {
+		if err := compHandler.CreateOrUpdateOrDelete(ctx, component, nil); err != nil {
 			r.SetDegraded("Error creating / updating resource", err, reqLogger)
 			return reconcile.Result{}, err
 		}
 	}
 
+	// Special handling for Typha and Node to deal with dependencies...
+	if err := compHandler.CreateOrUpdateOrDelete(ctx, typhaComponent, nil, utils.WithPreUpdateGate(r.shouldSkipTyphaUpdate)); err != nil {
+		r.SetDegraded("Error creating / updating resource", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+	if err := compHandler.CreateOrUpdateOrDelete(ctx, nodeComponent, nil, utils.WithPreUpdateGate(r.shouldSkipNodeUpdate)); err != nil {
+		r.SetDegraded("Error creating / updating resource", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
 	// TODO: We handle too many components in this controller at the moment. Once we are done consolidating,
 	// we can have the CreateOrUpdate logic handle this for us.
-	r.status.AddDaemonsets([]types.NamespacedName{{Name: "calico-node", Namespace: "calico-system"}})
-	r.status.AddDeployments([]types.NamespacedName{{Name: "calico-kube-controllers", Namespace: "calico-system"}})
+	r.status.AddDaemonsets([]types.NamespacedName{nodeNN})
+	r.status.AddDeployments([]types.NamespacedName{typhaNN})
```

Comment:
Typha seemed to be missing; not sure if that was deliberate but I added it here!